utils/sccli.py

```diff
@@ -55,6 +57,48 @@ def system(cmd):
     returncode = ret.returncode
     return out, err, returncode
 
+def get_all_interface_ip():
+    interface_list = ''
+    for interface in netifaces.interfaces():
+        addr = netifaces.ifaddresses(interface).get(netifaces.AF_INET)
+        if addr:
+            interface_list += "%s," % addr[0].get('addr')
+    return interface_list
+
+def set_proxy():
+    proxy = os.getenv('PROXY', '')
+    if proxy:
+        interface_ip_list = get_all_interface_ip()
+        predefine_no_proxy_list = ".xip.io,172.30.0.0/16,172.17.0.0/16"
+        proxy_user = quote_plus(os.getenv('PROXY_USER',''))
+        if proxy_user:
+            proxy_password = quote_plus(os.getenv('PROXY_PASSWORD',''))
+            http_proxy_url = "http://%s:%s@%s" % (proxy_user, proxy_password, proxy)
+            https_proxy_url = "https://%s:%s@%s" % (proxy_user, proxy_password, proxy)
+        else:
+            http_proxy_url = "http://%s" % proxy
+            https_proxy_url = "https://%s" % proxy
+        # openshift proxy setup
+        if system(('sed -i -e "/^#HTTP_PROXY=*/cHTTP_PROXY=%s"'
+                   ' -e "/^#HTTPS_PROXY=*/cHTTPS_PROXY=%s"'
+                   ' -e "/^#NO_PROXY=*/cNO_PROXY=%s%s"'
+                ' %s') % (http_proxy_url, https_proxy_url, interface_ip_list,
+                          predefine_no_proxy_list, OPENSHIFT_OPTION))[2]:
+            return ("Permisison denined: %s" % OPENSHIFT_OPTION)
+        # docker daemon proxy setup
+        if not os.path.isdir('/etc/systemd/system/docker.service.d'):
+            subprocess.call("mkdir /etc/systemd/system/docker.service.d", shell=True)
+        env_file_content = ('[Service]\n'
+                'Environment="HTTP_PROXY=%s" "HTTPS_PROXY=%s" "NO_PROXY=localhost,127.0.0.1,::1,.xip.io"\n') \
+                        % (http_proxy_url, https_proxy_url)
+        try:
+            with open('/etc/systemd/system/docker.service.d/http-proxy.conf', 'w') as fh:
+                fh.write(env_file_content)
```

Comment:
Can we separate setting proxy for docker environment and  OpenShift in two different functions and we can call the function for docker proxy in function for OpenShift. Also we need to add separate CLI i.e. interface for adding proxy for docker and OpenShift.